raiden/transfer/events.py

```diff
@@ -427,6 +427,11 @@ class SendProcessed(SendMessageEvent):
     pass
 
 
+@dataclass(frozen=True)
+class RequestMetadata(SendMessageEvent):
```

Comment:
Im wondering if `RequestMetadata` is semantically a `SendMessageEvent` since it will not send messages eventually. Have you already plans how to incorporate the fetched metadata afterwards?

Reply:
I'm trying to change this to be an `Event`, since this will not send messages to any other node. Maybe at this point of the PR is better to check the big picture, since this kind of detail is still changing.

I still don't know how I'll incorporate the fetched metadata, but my next step is to handle the event to fetch it. I'm trying to follow your advice to break this down in multiple steps.

Reply:
sounds good to me. So I think there is an alternative solution for it which might be a bit less complex. It might not be necessary to loop the fetched metadata back into the state machine. As we will probably only ever need the metadata upon handling Events (Which results in Raiden doing something with the outer world) we can fetch the Metadata and then instantly do whatever we wanted to do directly. 
This means that we dont have to go the whole way looping metadata back into the state machine.
You could see this like a "pre-event" which fetches the metadata and then once they are there we could handle the actual events which rely on the metadata. 

If you remember our discussion, we did not prefer this solution because we did not want fetching metadata to block other events which are handled with it but do not rely on the metadata. (Imagine we have a list of events to send multiple messages but for only one message metadata are missing, then we could actually handle all other events without waiting for the event).

I recently found out that the current code base already has a solution for such "blocking events". [Here](https://github.com/raiden-network/raiden/blob/7c539dd3d97077cec897121fc77bfe64379b5d63/raiden/raiden_service.py#L1014) you can add a blocking event. This results in that events of these types get handled in their own greenlet not blocking other "fast events". I think we could find a way where we would wrap "fast events" into a "missing metadata event" which is a blocking event. Then we could fetch metadata for all inner events and handle them after the metadata are fetched. 

Let me know what you think about it, we can also discuss later.

Reply:
Your current PR would actually fit very well for this type of solution

Reply:
I'll try to fit this strategy in my current solution then we can check how this will look like. I had already reached that part of blocking events that you mentioned. I'm trying to make it less complex than looping things back into the state machine but still keep it non-blocking.